model/auth_test.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2021 The Serverless Workflow Specification Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	val "github.com/serverlessworkflow/sdk-go/v2/validator"
+)
+
+func TestAuthDefinitionsStructLevelValidation(t *testing.T) {
+	type testCase struct {
+		desp     string
+		authDefs AuthDefinitions
+		err      string
+	}
+	testCases := []testCase{
+		{
+			desp: "nil defs",
+			authDefs: AuthDefinitions{
+				Defs: nil,
+			},
+			err: ``,
+		},
+		{
+			desp: "zero length defs",
+			authDefs: AuthDefinitions{
+				Defs: []Auth{},
+			},
+			err: ``,
+		},
+		{
+			desp: "multi unique defs",
+			authDefs: AuthDefinitions{
+				Defs: []Auth{
+					{
+						Name: "1",
+					},
+					{
+						Name: "2",
+					},
+					{
+						Name: "3",
+					},
+				},
+			},
+			err: ``,
+		},
+		{
+			desp: "multi non-unique defs",
+			authDefs: AuthDefinitions{
+				Defs: []Auth{
+					{
+						Name: "1",
+					},
+					{
+						Name: "2",
+					},
+					{
+						Name: "1",
+					},
+				},
+			},
+			err: `Key: 'AuthDefinitions.Name' Error:Field validation for 'Name' failed on the 'reqnameunique' tag`,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.desp, func(t *testing.T) {
+			g := NewWithT(t)
+
+			err := val.GetValidator().Struct(tc.authDefs)
+			if tc.err != "" {
+				g.Expect(err).To(HaveOccurred())
+				g.Expect(err.Error()).To(MatchRegexp(tc.err))
```

Comment:
Despite liking gomega for more assertion features, I believe we can have the same with testify, no? Not sure if adding one more dependency for this is a reasonable tradeoff.

Reply:
I prefer use gomega instead of testify because it's more powerful, and we should only keep one assert pkg for unittest.    If you have no opinion I can switch all the unit tests to gomega, how do you think？

Reply:
@lsytj0413, I understand that gomega is more powerful, though I don't think the tradeoff is worth it for our small use cases. However, we can keep an eye on it and switch in the future if needed. Can you please switch to testify? Many thanks!

Reply:
Ok, i have been switch gomega to testify, please take a look at it.